### DISCUSSION TITLE
Why are mikroorm devDependencies installed when calling yarn install --production ?

### DISCUSSION THREAD

**Discussion by Axedyson** [mikro-orm/mikro-orm#3322] [Q&A] [answered]
Here are my npm package dependencies:
```JSON
  "dependencies": {
    "@mikro-orm/core": "^5.2.3",
    "@mikro-orm/reflection": "^5.2.3",
    "apollo-server-core": "^3.10.0",
    "apollo-server-express": "^3.10.0",
    "argon2": "^0.28.7",
    "class-validator": "^0.13.2",
    "connect-redis": "^6.1.3",
    "express": "^4.18.1",
    "express-session": "^1.17.3",
    "graphql": "^15.8.0",
    "graphql-scalars": "^1.17.0",
    "ioredis": "^5.2.1",
    "reflect-metadata": "^0.1.13",
    "type-graphql": "^1.1.1"
  },
  "devDependencies": {
    "@mikro-orm/cli": "^5.2.3",
    "@mikro-orm/postgresql": "^5.2.3",
    "@mikro-orm/seeder": "^5.2.3",
    "@types/connect-redis": "^0.0.18",
    "@types/express": "^4.17.13",
    "@types/express-session": "^1.17.5",
    "@types/jest": "^28.1.6",
    "@types/node": "^18.0.6",
    "@types/supertest": "^2.0.12",
    "jest": "^28.1.3",
    "supertest": "^6.2.4",
    "ts-jest": "^28.0.7",
    "ts-node-dev": "^2.0.0",
    "typescript": "^4.7.4"
  }
```
As you can see the following packages:
```JSON
    "@mikro-orm/cli": "^5.2.3",
    "@mikro-orm/postgresql": "^5.2.3",
    "@mikro-orm/seeder": "^5.2.3",
```
are declared as devDependencies. When I run `yarn install --production` they are still installed when I look in the node_modules folder but other devDependencies such as `ts-node-dev` are not installed as expected. Why is that? I would also expect the mikro-orm devDependencies to not be installed. Here is an image of the unexpected installed dependencies that I found:
<img width="388" alt="image" src="https://user-images.githubusercontent.com/28491857/180417744-fee0378a-ab91-42d6-aaf7-96b03c22288b.png">

**Comment by B4nan** [comment, marked as answer]
I guess you would be better of asking at yarn repository instead. Maybe @merceyz will have some ideas? 

Btw why do you have the driver as dev dependency? That does not make much sense to me, you cant really use the core package without any driver.

**Reply by Axedyson** [reply]
Thanks for the fast reply! Yeah I was just experimenting a bit, I've moved the `@mikro-orm/postgresql` from devDependencies to dependencies now, thanks

**Reply by Axedyson** [reply]
By the way I'm also running all of this inside a dockerfile using `node:16.15.1-alpine3.16` as a base image. This image only includes yarn v1 not v2, so maybe that could be the culprit

**Reply by Axedyson** [reply]
Okay it seems like the only unexpected package that get's installed when using `yarn installed --production` is the `seeder` package, which I've declared as a devDependency

**Reply by B4nan** [reply]
Have you tried `yarn why`?

**Reply by Axedyson** [reply]
This is what I get when I say `yarn why`, note that I'm using yarn v1 here:
<img width="476" alt="image" src="https://user-images.githubusercontent.com/28491857/180478539-c39067bd-0780-4a50-b202-5efa29e3705c.png">

**Reply by Axedyson** [reply]
It's weird that it also finds `ts-node-dev`. This is what I get when I remove the line with `ts-node-dev` from the package.json file (not using yarn remove):
<img width="315" alt="image" src="https://user-images.githubusercontent.com/28491857/180479151-07e56d2c-5de9-4221-a0a9-0dd845c771af.png">
Here is also `yarn why` for `@mikro-orm/core`:
<img width="476" alt="image" src="https://user-images.githubusercontent.com/28491857/180479571-2e6413a3-db05-4f92-89b7-3d0cc898b6b2.png">

**Reply by merceyz** [reply]
> Maybe @merceyz will have some ideas?

Looks like a bug in Yarn v1, v3 behaves as expected.
